spec/requests/candidate_interface/authentication_spec.rb

```diff
@@ -0,0 +1,37 @@
+require_relative '../../../app/services/lib/magic_link_token'
+require 'rails_helper'
+
+RSpec.describe 'Authentication for candidates', type: :request do
+  it 'redirects the user if the token does not exist' do
+    get candidate_interface_application_form_url(token: '123')
+
+    # TODO: add a better check once we have implemented error messages
+    expect(response).to have_http_status(302)
+  end
+
+  it 'redirects the user if the token is missing' do
```

Comment:
'... token is missing'  and 'token does not exist' (from the previous test) sounds very similar, you may want to make it clearer.
I think you want to say:
- the token param is missing
- maybe the second means that token is invalid (hence does not exist)